src/sidebar/oauth-auth.js

```diff
@@ -86,6 +86,59 @@ function auth($http, $window, flash, random, settings) {
     return $http.post(tokenUrl, data, requestConfig);
   }
 
+  function grantTokenFromHostPage() {
+    var cfg = serviceConfig(settings);
+    if (!cfg) {
+      return null;
+    }
+    return cfg.grantToken;
+  }
+
+  /**
+   * Return the storage key used for storing access/refresh token data for a given
+   * annotation service.
+   */
+  function storageKey() {
+    // Use a unique key per annotation service. Currently OAuth tokens are only
+    // persisted for the default annotation service. If in future we support
+    // logging into other services from the client, this function will need to
+    // take the API URL as an argument.
+    var apiDomain = new URL(settings.apiUrl).hostname;
```

Comment:
Taking the hostname prevents a single server from serving two different annotation services at `https://example.com/service1/api` and `https://example.com/service2/api`. I guess that's not a huge deal, but it does seem unnecessary, why not just use the whole `apiUrl`?